### PR TITLE
Add helper account lifetime statistic, provide data for requested matches with priority

### PR DESCRIPTION
**Pull request by realmayus**

### CODE REVIEW

File: graphql/statistics/fields.ts [hunk 1]
```diff
@@ -425,6 +425,98 @@ export class StatisticsResolver {
...
+                from: 300 * 24 * 3600 * 1000,
+                to: -1,
+                value: 0,
+                label: '300+ Tage',
+            },
+        ];
+        lifetimes.forEach(({ student_id, lifetime }) => {
+            buckets.find((b) => b.from <= lifetime && (b.to > lifetime || b.to === -1)).value += 1;
+        });
+    }
```
Comment (Jonasdoubleyou) [inline graphql/statistics/fields.ts:518]: return buckets?

File: graphql/statistics/fields.ts [hunk 2]
```diff
@@ -425,6 +425,98 @@ export class StatisticsResolver {
         });
     }
 
+    @FieldResolver(() => [Bucket])
+    @Authorized(Role.ADMIN)
+    async tutorLifetime(@Root() statistics: Statistics, @Arg('onlyInactive') onlyInactive: boolean) {
```
Comment (Jonasdoubleyou) [inline graphql/statistics/fields.ts:430]: This is a pretty heavy query, where do we plan to integrate it?

File: graphql/statistics/fields.ts [hunk 3]
```diff
@@ -425,6 +425,98 @@ export class StatisticsResolver {
...
+                ))
+                GROUP BY
+                    student.id
+            )
+            SELECT
+                last_action.student_id,
+                EXTRACT(EPOCH FROM (last_action.last_action - first_action.first_screening)) as lifetime
+            FROM first_action
+            JOIN last_action ON first_action.student_id = last_action.student_id
+            WHERE first_action >= ${statistics.from}::timestamp
```
[the remark this answers is not in the document]
Comment (realmayus) [reply graphql/statistics/fields.ts:473]: Oops, I added this after testing the main part of the query 😅

File: graphql/statistics/fields.ts [hunk 4]
```diff
@@ -425,6 +425,98 @@ export class StatisticsResolver {
...
+                WHERE student.active = TRUE AND student."isStudent" = TRUE
+                AND (${onlyInactive} = false OR (
+                    NOT EXISTS (
+                        SELECT 1 FROM match WHERE match."studentId" = student.id AND match.dissolved = FALSE
+                    ) AND
+                    NOT EXISTS (
+                        SELECT 1 FROM subcourse
+                        LEFT JOIN subcourse_instructors_student instructor ON instructor."subcourseId" = subcourse.id
+                        LEFT JOIN lecture ON lecture."subcourseId" = subcourse.id
+                        WHERE instructor."studentId" = student.id AND lecture.start > ${statistics.to}::timestamp
```
[the remark this answers is not in the document]
Comment (realmayus) [reply graphql/statistics/fields.ts:462]: This is supposed to only count accounts that are not active anymore if the flag is set, which apparently means "no active match, no active course"

File: graphql/statistics/fields.ts (hunk 2 above)
Comment (realmayus) [reply graphql/statistics/fields.ts:430]: It's supposed to be added to the 1:1 tutoring statistics dashboard on Retool, below "Aktive Zusammenarbeit" (same visualization with bucketed data)

**Review by Jonasdoubleyou — APPROVED** [review]

**Pull request merged**